smartsim/entity/model.py

```diff
@@ -56,132 +48,151 @@
+    ) -> None:
+        """Initialize an ``Application``
 
         :param name: name of the application
         :param exe: executable to run
         :param exe_args: executable arguments
-        :param params: application parameters for writing into configuration files or
-                       to be passed as command line arguments to executable.
-        :param run_settings: launcher settings specified in the experiment
-        :param params_as_args: list of parameters which have to be
-                               interpreted as command line arguments to
-                               be added to run_settings
-        :param batch_settings: Launcher settings for running the individual
-                               application as a batch job
-        :param files: Files to have available to the application
+        :param files: files to be copied, symlinked, and/or configured prior to
+                      execution
+        :param file_parameters: parameters and values to be used when configuring
+                                files
         """
-        super().__init__(name, run_settings)
-        self.exe = [expand_exe_path(exe)]
-        # self.exe = [exe] if run_settings.container else [expand_exe_path(exe)]
-        self.exe_args = exe_args or []
-        self.params = params.copy() if params else {}
-        self.params_as_args = params_as_args
-        self.incoming_entities: t.List[SmartSimEntity] = []
+        super().__init__(name)
+        self._exe = [expand_exe_path(exe)]
+        self._exe_args = self._build_exe_args(exe_args) or []
+        self._files = copy.deepcopy(files) if files else None
+        self._file_parameters = (
+            copy.deepcopy(file_parameters) if file_parameters else {}
+        )
+        self._incoming_entities: t.List[SmartSimEntity] = []
         self._key_prefixing_enabled = False
-        self.batch_settings = batch_settings
-        self._fs_models: t.List[FSModel] = []
-        self._fs_scripts: t.List[FSScript] = []
-        self.files = copy.deepcopy(files) if files else None
+
+    # TODO Talk through as a group if _key_prefixing_enabled
+    #     should have proeprty and setter decorators or do we stick with something of similar syntax.
+    #     Bring this up to the group after the rest of the class is done so they see what a consistent
+    #     API is currently being formed.
+    # TODO Discuss if the exe_args parameter should be set with a str in the construct
+    #     and setter or should we stick to t.Sequence[str] only.  This might require group discussion.
+    # TODO Discuss with the core team when/if properties shoulda always be returned via reference
+    #     or deep copy
+    # TODO Ticket says to remove prefixing, but I think that needs to stay
+    # TODO Should attached_files_table be deleted and replaced with @property?
+    # TODO Put create pinning string into a new ticket for finding a home for it
```

Comment:
Looks like both of these are done! Feel free to rm!!

Reply:
removed